packages/flutter/lib/src/material/button_theme.dart

```diff
@@ -222,7 +224,7 @@ class ButtonTheme extends InheritedTheme {
       buttonTheme ??= theme.buttonTheme;
       if (buttonTheme.colorScheme == null) {
```

Comment:
If you're requiring that `colorScheme` be defined in the constructors, this condition shouldn't be necessary. That being said, it might be worth leaving in the assert in line 229 to prevent unintended behavior

Reply:
I can't comment on it, but line 222 should also probably be updated to simply check if `buttonTheme` is null. Let me know if there's something I'm missing

Reply:
Did the suggested changes.